libgit/browser.go

```diff
@@ -228,9 +236,23 @@ func (b *Browser) Join(elem ...string) string {
 
 // ReadDir implements the billy.Filesystem interface for Browser.
 func (b *Browser) ReadDir(p string) (fis []os.FileInfo, err error) {
+	if p == "" {
+		p = "."
+	}
+
+	b.lock.RLock()
```

Comment:
All this locking is ridiculous.  Would be much clearer with a helper function that returns a bool if the returned list should be used or not.

Reply:
> ridiculous

That's what "R" in RLock stands for!